File: help_reviews.client/src/models/Restaurant.js
import { Profile } from "./Account.js"

export class Restaurant {
  constructor (data) {
    this.id = data.id
    this.name = data.name
    this.imgUrl = data.imgUrl
    this.description = data.description
    this.isShutdown = data.isShutdown
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.reportCount = data.reportCount
  }
}

File: help_reviews/Models/Restaurant.cs
namespace help_reviews.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  public string ImgUrl { get; set; }

  public string Description { get; set; }
  public bool? IsShutdown { get; set; }
  public string CreatorId { get; set; }
  public int reportCount { get; set; } = 0;
  public Profile Creator { get; set; }
}
